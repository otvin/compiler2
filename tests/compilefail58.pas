program compilefail58(output);
{an array with index types of type a and b, should error if someone tries to reference it via [typeb, typea]}

type
	subrint = 1..10;
	subrchar = 'a'..'z';
	myarr = array [subrint] of array [subrchar] of boolean;

var
	i:integer;
	c:char;
	a:myarr;

begin

	i := 5;
	c := 'a';
	a[c, i] := True;

	write(a[c,i]);
end.
