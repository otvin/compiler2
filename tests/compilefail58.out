Incorrect type for array index in file:tests/compilefail58.pas line:18    column:7  
